DOCS-5916 adding contracts (#87)
mulesoft/docs-exchange: modules/ROOT/pages/about-my-applications.adoc

image::ex2-my-applications.png[]

A contract holds information including a client ID and optional client secret
for the API. Each application lists a description, its URL, redirect URIs,

> grant types, and a usage dashboard.

Reviewer: Does the contract hold information FOR the API or ABOUT the API?
Would it be just as accurate to say:
"A contract includes information about an API, including its client ID; its client secret, if any; its description; its URIs; its grant types, and its usage dashboard"?
Or are these two sentences about two different things: the contract that includes ID and secret, and the API that includes the description etc.?
Clarify.

Reviewer 2: A contract is the link between a client application and an API. An API can have only one contract with a specific client application. A client application can have several contracts with different APIs. The only thing a contract may hold is SLA tier info and terms and conditions. A client application is identified by a client ID and client secret (both mandatory and generated by the platform). A client application holds as well, description , grant types supported, etc. Usage (aka analytics) is stored in a different place.

The usage information enables you to set the verification duration. If not specified,
the range is the previous eight days. The usage lists the total number of requests to
the API, average latency, the error rate, and graphs for total requests,

> requests by HTTP status code, and the error percentage.

Reviewer: the range is the previous eight days. The usage lists total number of requests to
the API, average latency, error rate, and graphs for total requests,
requests by HTTP status code, and error percentage.

Reviewer: Why is each line its own paragraph (+)? 
I see this method, and then I also see a method in which the + appears only at the beginning of a new paragraph.
Would you mind sending me a separate Slack message or email to explain?

Reviewer 2: Please see: https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#breaks

A contract holds information about the contract with a client ID and client secret for the API.
Each application lists a description, its URL, redirect URIs, grant types, and a usage dashboard.

> The usage information lets you set the verification duration. If not specified,

Reviewer 3: ```suggestion
The usage information enables you to set the verification duration. If not specified,
```

> the range is the last 8 days. The usage lists the total number of requests to

Reviewer 3: ```suggestion
the range is the previous eight days. The usage lists the total number of requests to
```

the API, average latency, the error rate, and graphs for total requests,

> average latency, requests by HTTP status code, and the error percentage.

Reviewer 3: ```suggestion
requests by HTTP status code, and the error percentage.
```

Reviewer 3: (average latency was repeated)

> The applications result when users click Request Access to establish

Reviewer 3: applications result? The application is created? I don't know what "applications result" means. 
If you don't have an application to link to your API, you can create the application by clicking the *Request Access* menu. You can then click *Request Access* to establish a link between your application and API.

Reviewer 3: That's my best guess. I'm not sure what's going on there.

Author: I rewrote for clarity, and included this text. There was a lot of sloppy text in this one, including things I missed such as "average latency" being repeated. Thank you for a very helpful review!

a link between an application and an API. Either the application already exists,
or the user creates the application from the Request Access menu.

Applications in My Applications are registered using API Manager in Anypoint Platform.

> Requesting access registers a client and creates an API contract between an API

Reviewer 3: ```suggestion
When you request access, a client is registered and creates an API contract between an API
```

> and a client application.

Reviewer 3: ```suggestion
and the client application.
```

> APIs that have been protected by a client ID enforcement policy require

Reviewer 3: ```suggestion
APIs that are protected by a client ID enforcement policy require
```

A contract holds information about the contract with a client ID and client secret for the API.
Each application lists a description, its URL, redirect URIs, grant types, and a usage dashboard.

The usage information enables you to set the verification duration. If not specified,
the range is the previous eight days. The usage lists the total number of requests to
the API, average latency, the error rate, and graphs for total requests,
requests by HTTP status code, and the error percentage.

Applications in My Applications are registered using API Manager in Anypoint Platform.

> To implement an API, the client application must first request access. When the

Reviewer 2: Incorrect statement. API can be created regardless of having one, many or no contracts. I would say instead of "To implement an API" something like "To register a client application to an existing API".

Author: Fixed. Thank you!

> request is approved by the API, a contract is created between the client

Reviewer 2: "request is approved by the API owner"

Author: Fixed.

application and the API, and the client application is registered.

If you don't have an application to link to your API, you can create the application
in the Request Access menu. You can then click Request Access to connect your

> application and the API.

Reviewer 2: This is a basic case, that can be all right. For your radar: If at least one SLA is defined in the API, you have to choose one when requesting access. Also, terms and conditions have to be accepted, if defined.

Author: This is good info. Added.

APIs that are protected by a client ID enforcement policy require client

> applications to provide a client ID and client secret. The client ID and client

Reviewer 2: client secret is optional

Author: Fixed.

secret credentials are automatically created when the client application is
registered. These credentials enable the application to access the API and
provide consumer analytics for an API developer.

> API contracts are visible to API consumers in Exchange, and visible to API

Reviewer 2: Not sure if this is true in general. If your are org owner, then yes. I believe (please double check) that only org owners and API owners have visibility over the contracts of a certain API.

Author: @nahuel-dallavecchia who can confirm this? Exchange developers?

Reviewer 2: Try with Samuel Centurion, QA Lead of API Manager first.

Author: Thank you, Nahuel! Looks like Sebastian Lamelas on the Exchange team already answered this one: "The API owner has visibility over API contracts in API Manager (visibility of ALL contracts ever created for that API instance), whereas the API consumer has visibility only of the API contracts created between an API instance and *his* application in "My Applications" page in Exchange."

Author: Fixed.
